### PR TITLE
feat(*): Adds `wash spy` command with experimental flag support

### LINKED ISSUE #599: [FEATURE] `wash spy`
To help users debug and understand their applications, we need an easy way for them to see what is going on. We should add a command to wash that allows a user to specify an actor (not by ID) and then wash automatically subs on the correct invocation topics and decodes msgpack for them

It could look kinda like this

```
$ wash spy echo
Listening for echo invocations

[TIMESTAMP] { method: "GET", body: [data] ...}
```

As part of this, we should add an "experimental" env var and flag that let's people turn on experiments like this one. This feels like an "easy win" feature but locks it behind a flag in case we change or remove it

### PR DESCRIPTION
**Pull request by thomastaylor312**
## Feature or Problem

Adds a helpful debugging tool that allows you to observe what is being called between your actors and providers

## Related Issues

Closes #599 

## Release Information
0.18? (If it gets reviewed fast enough) 0.19 otherwise

## Consumer Impact

None, new experimental feature

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

### Unit Test(s)/Integration

None. To actually test this you need a full setup of actors and then have to test output. Since this is experimental, I figure the manual testing is fine

### Manual Verification

I tried spinning up several actors and providers (both provider -> actor and actor -> provider) to test that we could get messages

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -224,6 +239,13 @@ async fn main() {
         CliCommand::RegPull(reg_pull_cli) => {
             common::registry_cmd::registry_pull(reg_pull_cli, output_kind).await
         }
+        CliCommand::Spy(spy_cli) => {
+            if !cli.experimental {
```
Comment (thomastaylor312) [inline src/main.rs:243]: Please note: This is a poor man's implementation of experimental. I was digging through all the clap docs and couldn't find a nice way to have clap do this at a `Command` level

File: crates/wash-lib/src/cli/spy.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    let ctl_client = wco.clone().into_ctl_client(None).await?;
+    let nats_client = wco.into_nats_client().await?;
+
+    let mut spier = Spier::new(&cmd.actor, &ctl_client, &nats_client).await?;
+
+    println!("Spying on actor {}\n", spier.actor_id());
+
+    while let Some(msg) = spier.next().await {
+        println!(
+            "[{}]\nFrom: {:<20}\tTo: {:<20}\tHost: {}\n\nOperation: {}\nMessage: {}\n",
```
Comment (connorsmith256) [inline crates/wash-lib/src/cli/spy.rs]: Supernit: consider `r#""` to drop the escape chars and make this more readable

File: crates/wash-lib/src/spier.rs [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        // Try parsing with msgpack and then with cbor. If neither work, then just return the raw
+        // NOTE(thomastaylor312): I don't think anyone else does their own encoding, but if that
+        // becomes popular, we can add support for it here
+        let mut serializer = serde_json::Serializer::pretty(BufWriter::new(Vec::new()));
+        let parsed = match serde_transcode::transcode(
+            &mut rmp_serde::Deserializer::new(&data[..]),
+            &mut serializer,
+        ) {
+            // SAFETY: We know that JSON writes to valid UTF-8 and that we won't fail flushing to a Vec
+            Ok(_) => String::from_utf8(serializer.into_inner().into_inner().unwrap()).unwrap(),
```
Comment (connorsmith256) [inline crates/wash-lib/src/spier.rs]: Ew... Not asking for a change, just... ew.

File: crates/wash-lib/src/spier.rs [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                &prov.link_name
+            );
+            nats_client.subscribe(topic)
+        }))
+        .await
+        .into_iter()
+        .collect::<Result<Vec<_>, _>>()?;
+        subs.push(actor_stream);
+
+        // TODO: Don't forget to filter provider messages by actor ID
```
Comment (connorsmith256) [inline crates/wash-lib/src/spier.rs]: This is done, right?

File: crates/wash-lib/src/spier.rs [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+                        // Just skip it if we can't parse it. This means we need to tell the executor to automatically wake up and poll immediately if we skip
+                        cx.waker().wake_by_ref();
+                        return Poll::Pending;
+                    }
+                };
+                let body = inv.msg;
+                inv.msg = Vec::new();
+
+                if inv.origin.is_provider() && inv.target.public_key() != self.actor_id.as_ref() {
+                    println!("Not for us: {:?}", inv);
```
Comment (connorsmith256) [inline crates/wash-lib/src/spier.rs]: Remove me?

File: crates/wash-lib/src/spier.rs (hunk 3 above)
Comment (thomastaylor312) [reply crates/wash-lib/src/spier.rs]: There is probably a better way, but 🤷

File: crates/wash-lib/src/spier.rs (hunk 4 above)
Comment (thomastaylor312) [reply crates/wash-lib/src/spier.rs]: Yes, just removed it

File: crates/wash-lib/src/spier.rs (hunk 3 above)
Comment (thomastaylor312) [reply crates/wash-lib/src/spier.rs]: I cleaned this up a bit by just using a Vec instead

**Review by connorsmith256 — APPROVED** [review]

**Pull request merged**